src/wallet/wallet.cpp

```diff
@@ -981,6 +981,45 @@ bool CWallet::LoadToWallet(const uint256& hash, const UpdateWalletTxFn& fill_wtx
     return true;
 }
 
+void CWallet::RemoveTransactions(const CScript& script)
+{
+    LOCK(cs_wallet);
+
+    std::vector<std::unique_ptr<CWalletTx>> vWtx;
+    for (std::map<uint256, CWalletTx>::iterator it = mapWallet.begin(); it != mapWallet.end(); ++it) {
+        const CWalletTx& wTx = it->second;
+        bool txContainsScript = false;
+
+        // if the tx sends to this script
+        for (const CTxOut& txout : wTx.tx->vout) {
+            txContainsScript |= script == txout.scriptPubKey;
+        }
+
+        // if the tx spends from this script
+        if (!txContainsScript) {
+            for (const CTxIn& txin : wTx.tx->vin) {
+                std::map<uint256, CWalletTx>::const_iterator mi = mapWallet.find(txin.prevout.hash);
+                if (mi != mapWallet.end()) {
+                    const CWalletTx& prev = (*mi).second;
+                    if (txin.prevout.n < prev.tx->vout.size()) {
+                        txContainsScript |= script == prev.tx->vout[txin.prevout.n].scriptPubKey;
+                    }
+                }
+            }
+        }
+
+        if (txContainsScript && !IsFromMe(*wTx.tx) && !IsMine(*wTx.tx)) {
+            vWtx.push_back(MakeUnique<CWalletTx>(this, MakeTransactionRef(std::move(*wTx.tx))));
```

Comment:
Please [use `std::make_unique` in new code](https://github.com/bitcoin/bitcoin/blob/master/src/util/memory.h#L13).